services/index/src/api.js

```diff
@@ -57,35 +54,31 @@ builder.declare({
   let namespace = indexPath.join('.');
 
   // Load indexed task
-  let tasks;
   try {
-    tasks = await this.IndexedTask.query({
-      namespace: namespace,
-      name: name,
-      expires: Entity.op.greaterThan(new Date()),
-    });
+    const { continuationToken: _, rows } = await helpers.taskUtils.getIndexedTasks(
+      this.db,
+      { namespace, name },
+      { query: req.query },
+    );
+    if (!rows.length) {
+      return res.reportError('ResourceNotFound', 'Indexed task not found', {});
+    }
+    const task = rows[0];
+    return res.reply(helpers.taskUtils.serialize(task));
   } catch (err) {
     // Re-throw the error, if it's not a 404
     if (err.code !== 'ResourceNotFound') {
```

Comment:
I don't think anything in `getIndexedTasks` would throw this error -- and the not-found case is handled above (`if (!rows.length)`).